### PR TITLE
raftstore: fix false stale peer alert

### PR DESCRIPTION
**Pull request by overvenus**
Signed-off-by: Neil Shen <overvenus@gmail.com>

## What have you changed? (mandatory)

Fix false stale peer alert.

## What are the type of the changes? (mandatory)

- Bug fix (non-breaking change which fixes an issue)

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No.

## Does this PR affect tidb-ansible update? (mandatory)

No.

## Refer to a related PR or issue link (optional)

#3384

### CODE REVIEW

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -752,6 +752,11 @@ impl Peer {
     }
 
     pub fn check_stale_state(&mut self) -> StaleState {
+        if self.is_leader() {
+            // Leaders always have valid state.
+            self.leader_missing_time = None;
```
Comment (siddontang) [inline src/raftstore/store/peer.rs]: need test to cover it?

Comment (overvenus) [reply src/raftstore/store/peer.rs]: It is hard to test, it only prints log when the bug is triggered.

Comment (siddontang) [reply src/raftstore/store/peer.rs]: em, maybe you can use fail-rs?

**Comment by overvenus** [comment]
PTAL @Hoverbear @siddontang

File: src/raftstore/store/fsm/peer.rs [hunk 2]
```diff
@@ -1955,7 +1955,9 @@ impl<T: Transport, C: PdClient> Store<T, C> {
             // In this case, peer B would notice that the leader is missing for a long time,
             // and it would check with pd to confirm whether it's still a member of the cluster.
             // If not, it destroys itself as a stale peer which is removed out already.
-            match peer.check_stale_state() {
+            let state = peer.check_stale_state();
+            fail_point!("peer_check_stale_state", state != StaleState::Valid, |_| {});
```
Comment (huachaohuang) [inline src/raftstore/store/fsm/peer.rs]: Can you explain why the state can be invalid for a leader before the fix?

**Pull request merged**